### ISSUE TITLE
Error during 4x4090GPU parallel training with unmodified source code

### ISSUE THREAD

**Issue by zc2023** [filaPro/oneformer3d#103]
Description:
Hello,
Thank you for your excellent work on this project. I recently attempted to train the model using 4 NVIDIA RTX 4090 GPUs in parallel, but encountered repeated errors regarding the computational graph. I have not made any modifications to the source code. I would greatly appreciate your advice on how to resolve this issue.

Steps to Reproduce:

Set up the training environment on a machine with 4 NVIDIA RTX 4090 GPUs.
I added following lines in the `oneformer3d_1xb4_scannet200.py` to help debug:
```
model_wrapper_cfg=dict(
    type='MMDistributedDataParallel',
    find_unused_parameters=True,
    detect_anomalous_params=True)
```
dist_train.sh:
```
python -m torch.distributed.launch --nproc_per_node=$GPUS --master_port=${PORT} \
    $(dirname "$0")/train.py $CONFIG --launcher pytorch ${@:3}
```
Run the training script below:
```
CUDA_VISIBLE_DEVICES=0,1,2,3 PYTHONPATH=./ bash ./tools/dist_train.sh configs/oneformer3d_1xb4_scannet200.py 4
```
Expected Behavior:
The training should proceed without errors.

Observed Behavior:
The following error is repeatedly logged during training:

```
module.backbone.final.kernel with shape torch.Size([96, 96]) is not in the computational graph
module.backbone.final.bias with shape torch.Size([1, 96]) is not in the computational graph
...
```

Attachment:
![bug](https://github.com/user-attachments/assets/3a8839c0-d7c7-42f0-bfd7-4786baa327de)

Could you please advise on how to resolve this issue? Let me know if you need more information to debug.

**Comment by filaPro** [comment]
Hi, did you try the solution from #3? Even without `MMDistributedDataParallel`.

**Comment by zc2023** [comment]
Thanks for your quick response! I have changed 
```
model_wrapper_cfg=dict(
    type='MMDistributedDataParallel',
    find_unused_parameters=True,
    detect_anomalous_params=True)
```
in `oneformer3d_1xb4_scannet200.py` into the following line you suggested:
```
find_unused_parameters = True
```
and it solves the above problem for parallel training.

**Comment by zc2023** [comment]
> Hi, did you try the solution from #3? Even without `MMDistributedDataParallel`.



> Thanks for your quick response! I have changed
> 
> ```
> model_wrapper_cfg=dict(
>     type='MMDistributedDataParallel',
>     find_unused_parameters=True,
>     detect_anomalous_params=True)
> ```
> 
> in `oneformer3d_1xb4_scannet200.py` into the following line you suggested:
> 
> ```
> find_unused_parameters = True
> ```
> 
> and it solves the above problem for parallel training.

**Comment by forever-rz** [comment]
@zc2023 May I ask how accurate are your results with 4-card training? How was the learning rate adjusted?

**Comment by zifte** [comment]
> Description: Hello, Thank you for your excellent work on this project. I recently attempted to train the model using 4 NVIDIA RTX 4090 GPUs in parallel, but encountered repeated errors regarding the computational graph. I have not made any modifications to the source code. I would greatly appreciate your advice on how to resolve this issue.
> 
> Steps to Reproduce:
> 
> Set up the training environment on a machine with 4 NVIDIA RTX 4090 GPUs. I added following lines in the `oneformer3d_1xb4_scannet200.py` to help debug:
> 
> ```
> model_wrapper_cfg=dict(
>     type='MMDistributedDataParallel',
>     find_unused_parameters=True,
>     detect_anomalous_params=True)
> ```
> 
> dist_train.sh:
> 
> ```
> python -m torch.distributed.launch --nproc_per_node=$GPUS --master_port=${PORT} \
>     $(dirname "$0")/train.py $CONFIG --launcher pytorch ${@:3}
> ```
> 
> Run the training script below:
> 
> ```
> CUDA_VISIBLE_DEVICES=0,1,2,3 PYTHONPATH=./ bash ./tools/dist_train.sh configs/oneformer3d_1xb4_scannet200.py 4
> ```
> 
> Expected Behavior: The training should proceed without errors.
> 
> Observed Behavior: The following error is repeatedly logged during training:
> 
> ```
> module.backbone.final.kernel with shape torch.Size([96, 96]) is not in the computational graph
> module.backbone.final.bias with shape torch.Size([1, 96]) is not in the computational graph
> ...
> ```
> 
> Attachment: ![bug](https://github.com/user-attachments/assets/3a8839c0-d7c7-42f0-bfd7-4786baa327de)
> 
> Could you please advise on how to resolve this issue? Let me know if you need more information to debug.

Hi, Could you please paste the whole dist_train.sh, I followed your codes as 
```
python -m torch.distributed.launch --nproc_per_node=$GPUS --master_port=${PORT} \
    $(dirname "$0")/train.py $CONFIG --launcher pytorch ${@:3}
```

but it still not worked. thanks
